proxy/mempool/mempool_schedule.py

```diff
@@ -489,7 +566,13 @@ def _check_oversized_and_reduce(self, new_tx: MPTxRequest) -> None:
     def peek_top_tx(self) -> Optional[MPTxRequest]:
         if len(self._sender_pool_queue) == 0:
             return None
-        return self._sender_pool_queue[self._top_index].top_tx
+
+        for sender_pool in reversed(self._sender_pool_queue):
+            top_tx = sender_pool.top_tx
+            if not self._tx_dict.is_gapped(top_tx):
+                return top_tx
```

Comment:
Let's fix the described problems, maybe it will resolve this issue.

Reply:
The reason I did this is because when `mempool.py` calls `acquire_tx()`, `self._sender_pool_queue` and `self._tx_dict` will be modified but there is no sync to rebalance the `sender_pool_queue` index.

I'm currently testing if putting a sync inside `acquire_tx()` will work, then I'll replace the filter `for` loop